Mathlib/Order/Partition/Finpartition.lean

```diff
@@ -586,6 +586,24 @@ theorem mem_part_ofSetoid_iff_rel {s : Setoid α} [DecidableRel s.r] {b : α} :
   simp only [← hc, mem_univ, mem_filter, true_and] at this ⊢
   exact ⟨s.trans (s.symm this), s.trans this⟩
 
+section Two
+
+/-- A bipartition defined by one part, the other part being its complement.
+Both parts must be non-empty. -/
+def bipartition (h1 : s ≠ ∅) (h2 : s ≠ univ) : Finpartition (univ : Finset α) where
```

Comment:
Why don't you define it as `indiscrete` if ever `s` is `∅` or `univ`?

Reply:
`α` might not be inhabited, in which case no finpartition at all can be defined on `univ : Finset α`.